core/src/validator.rs

```diff
@@ -1677,18 +1681,16 @@ fn active_vote_account_exists_in_bank(bank: &Bank, vote_account: &Pubkey) -> boo
     false
 }
 
-fn check_poh_speed(
-    genesis_config: &GenesisConfig,
-    maybe_hash_samples: Option<u64>,
-) -> Result<(), ValidatorError> {
-    if let Some(hashes_per_tick) = genesis_config.hashes_per_tick() {
-        let ticks_per_slot = genesis_config.ticks_per_slot();
+fn check_poh_speed(bank: &Bank, maybe_hash_samples: Option<u64>) -> Result<(), ValidatorError> {
+    if let Some(hashes_per_tick) = bank.hashes_per_tick() {
+        let ticks_per_slot = bank.ticks_per_slot();
         let hashes_per_slot = hashes_per_tick * ticks_per_slot;
         let hash_samples = maybe_hash_samples.unwrap_or(hashes_per_slot);
 
         let hash_time = compute_hash_time(hash_samples);
         let my_hashes_per_second = (hash_samples as f64 / hash_time.as_secs_f64()) as u64;
-        let target_slot_duration = Duration::from_nanos(genesis_config.ns_per_slot() as u64);
+
+        let target_slot_duration = Duration::from_nanos(bank.ns_per_slot as u64);
```

Comment:
Not really sure `ns_per_slot` should be public in `Bank`, but seemingly a refactor for after this:
https://github.com/anza-xyz/agave/blob/4e5af1e90b0efee2d06c6b6ea63adc063cda2b10/runtime/src/bank.rs#L730